ingestion_tools/dataset_configs/config-attribute-discovery.py

```diff
@@ -0,0 +1,213 @@
+"""
+
+
+def recursive_dict_update_list_helper(current_entries, key, new_entries):
+    # nothing to update with
+    if len(new_entries) == 0:
+        return current_entries
+
+    for i in range(len(new_entries)):
+        # if there are nested dictionaries, recurse as neccessary
+        if isinstance(new_entries[i], dict):
+            # dictionary entry that corresponds to the new_entries[i] dictionary (the value that all the new_entries dicts are getting merged into)
+            # default empty dict
+            corresponding_entry = {}
+            # if a corresponding value alrady exists:
+            if key in current_entries:
+                # if it's a list, then it should have been created by this function
+                # it is a list to represent the fact that it's a multivalued element (so that when the yaml gets generated, it is noted as a multivalued attribute)
+                # this is only reason why it's a list: inside the list it should just have the dictionary that is new_entries[i]'s corresponding value
+                if isinstance(current_entries[key], list):
+                    assert len(current_entries[key]) == 1
+                    # assign accordingly
+                    corresponding_entry = current_entries[key][0]
+                # if it's a dictionary, we just use it
+                # note that it will never be a dictionary again, because after updating this multivalued field (we know so because we have new_entries as list),
+                # it will become a list element with one dict element
+                elif isinstance(current_entries[key], dict):
+                    corresponding_entry = current_entries[key]
+                else:
+                    raise ValueError(
+                        f"Unresolvable conflict for {current_entries[key]} (type: {type(current_entries[key])}), {new_entries[i]} (type: {type(new_entries[i])})",
+                    )
+
+            current_entries[key] = recursive_dict_update(corresponding_entry, new_entries[i])
+        else:
+            raw_data_check(current_entries.get(key), new_entries)
+            current_entries[key] = new_entries
+    # if new_entries was a list of dictionaries, the current_entries[key] will be a dictionary instead of a list (recursive_dict_update returns a dictionary)
+    # so we need to convert it back to a list
+    if not isinstance(current_entries[key], list):
+        current_entries[key] = [current_entries[key]]
+    return current_entries
+
+
+"""
+This function is the main function that will be called. It will take in the current dictionary and the new dictionary, and will merge them together. A recursive
+approach is used to handle nested dictionaries and lists. The function will return the merged dictionary.
+"""
+
+
+def recursive_dict_update(current_entries, new_entries):
+    for key, new_value in new_entries.items():
+        # Regular scenarios
+        if new_value is None:
```

Comment:
We should still include fields that have the value set as None. As they could be an expected key in the ingestion processing.

Reply:
Noted! linkml may treat a value set as None as not existing (like the attribute was never provided in the config file), but I'll have to look into this more.